src/js/actions/documents.js

```diff
@@ -647,11 +665,14 @@ define(function (require, exports) {
                 return descriptor.playObject(documentLib.select(documentRef));
             })
             .then(function () {
-                var payload = {
-                    selectedDocumentID: document.id
-                };
-
-                this.dispatch(events.document.SELECT_DOCUMENT, payload);
+                if (!document.layers) {
+                    // The now-active document has yet to be fully initialized.
+                    return this.transfer(updateDocument);
+                } else {
+                    this.dispatch(events.document.SELECT_DOCUMENT, {
```

Comment:
1. No: `dispatch` is perfectly safe inside `then` handlers. 
2. No: `updateDocument`, when no ID is supplied, updates the _active_ document, which updates the application store accordingly.